plugins/providers/virtualbox/cap/configure_disks.rb

```diff
@@ -75,39 +124,75 @@ def self.get_current_disk(machine, disk, all_disks)
+            disk_metadata = create_disk(machine, disk, controller)
           elsif compare_disk_size(machine, disk, current_disk)
-            disk_metadata = resize_disk(machine, disk, current_disk)
+            disk_metadata = resize_disk(machine, disk, current_disk, controller)
           else
             # TODO: What if it needs to be resized?
 
             disk_info = machine.provider.driver.get_port_and_device(current_disk["UUID"])
             if disk_info.empty?
               LOGGER.warn("Disk '#{disk.name}' is not connected to guest '#{machine.name}', Vagrant will attempt to connect disk to guest")
-              dsk_info = get_next_port(machine)
-              machine.provider.driver.attach_disk(dsk_info[:port],
+              dsk_info = get_next_port(machine, controller)
+              machine.provider.driver.attach_disk(controller.name,
+                                                  dsk_info[:port],
                                                   dsk_info[:device],
+                                                  "hdd",
                                                   current_disk["Location"])
+              disk_metadata[:port] = dsk_info[:port]
+              disk_metadata[:device] = dsk_info[:device]
             else
               LOGGER.info("No further configuration required for disk '#{disk.name}'")
+              disk_metadata[:port] = disk_info[:port]
+              disk_metadata[:device] = disk_info[:device]
             end
 
-            disk_metadata = {uuid: current_disk["UUID"], name: disk.name}
+            disk_metadata[:uuid] = current_disk["UUID"]
+            disk_metadata[:name] = disk.name
+            disk_metadata[:controller] = controller.name
           end
 
           disk_metadata
         end
 
+        # Handles all disk configs of type `:dvd`
+        #
+        # @param [Vagrant::Machine] machine - the current machine
+        # @param [Config::Disk] dvd - the current disk to configure
+        # @param [VagrantPlugins::ProviderVirtualBox::Model::StorageController] controller -
+        # the storage controller to use
+        # @return [Hash] - dvd_metadata
+        def self.handle_configure_dvd(machine, dvd, controller)
+          disk_info = get_next_port(machine, controller)
+          port = disk_info[:port]
+          device = disk_info[:device]
+
+          machine.provider.driver.attach_disk(controller.name, port, device, "dvddrive", dvd.file)
+
+          # Refresh the controller information
+          storage_controllers = machine.provider.driver.read_storage_controllers
+          controller = storage_controllers.detect { |c| c.name == controller.name }
+          attachment = controller.attachments.detect { |a| a[:port] == port && a[:device] == device }
+
+          if attachment
+            { uuid: attachment[:uuid], name: dvd.name, controller: controller.name, port: port, device: device }
+          else
+            {}
+          end
```

Comment:
Hm, maybe we could be a bit more explicit here and define a `dvd_meta = {}` at the top. Then change this `if` to set `dvd_meta` to the hash above, then remove the else, and always return `dvd_meta` at the bottom.